cmd/plugin/node_server.go

```diff
@@ -25,20 +28,36 @@ const (
 	ctxKeyEphemeralVolume = "csi.storage.k8s.io/ephemeral"
 )
 
-func NewNodeServer(driver *csicommon.CSIDriver, mounter *backend.SnapshotMounter, imageSvc cri.ImageServiceClient, secretStore secret.Store) *NodeServer {
+type ImagePullStatus int
+
+func NewNodeServer(driver *csicommon.CSIDriver, mounter *backend.SnapshotMounter, imageSvc cri.ImageServiceClient, secretStore secret.Store, asyncImagePulls bool) *NodeServer {
 	return &NodeServer{
 		DefaultNodeServer: csicommon.NewDefaultNodeServer(driver),
 		mounter:           mounter,
-		imageSvc:          imageSvc,
 		secretStore:       secretStore,
+		asyncImagePulls:   asyncImagePulls,
+		mutex:             &sync.Mutex{},
```

Comment:
Q: where are we using the mutex ?

Reply:
Good call out. I will remove this. Leftover from before refactor.